plonky2/examples/fibonacci.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::Result;
+use plonky2::field::types::Field;
+use plonky2::iop::witness::PartialWitness;
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+use plonky2::plonk::circuit_data::CircuitConfig;
+use plonky2::plonk::config::{GenericConfig, PoseidonGoldilocksConfig};
+use plonky2::plonk::verifier::verify;
+
+fn main() -> Result<()> {
+    const D: usize = 2;
+    type C = PoseidonGoldilocksConfig;
+    type F = <C as GenericConfig<D>>::F;
+
+    let config = CircuitConfig::standard_recursion_config();
+
+    let pw = PartialWitness::new();
+    let mut builder = CircuitBuilder::<F, D>::new(config);
+
+    let mut prev_target = builder.zero();
+    let mut cur_target = builder.one();
+    for _ in 0..99 {
+        let temp = builder.add(prev_target, cur_target);
+        prev_target = cur_target;
+        cur_target = temp;
+    }
+
+    let fib_100 = F::from_canonical_u64(3736710860384812976);
+    let fib_100_target = builder.constant(fib_100);
+    builder.register_public_input(fib_100_target);
+
+    builder.connect(fib_100_target, cur_target);
```

Comment:
I think a more standard design would be to just register `cur_target` as a public input.
That way we keep the same statement "I know the 100th Fibonacci number" without having to hardcode the value in the circuit.

Reply:
Ah yes that makes sense

Reply:
Does it look right now?

Reply:
Also not quite sure how best to do that in `square_root`...

Reply:
> Does it look right now?

Yes looks good! You can also remove that part now:
```rust
let fib_100 = F::from_canonical_u64(3736710860384812976);
let fib_100_target = builder.constant(fib_100);
builder.connect(fib_100_target, cur_target);
```